### ISSUE TITLE
Concatenate multiple graphql files inside a directory to a single string file.

### ISSUE THREAD

**Issue by akashbdj** [gobuffalo/packr#167]
Hi @markbates ,

I've multiple `.graphql` files describing our schema. I want to pack all my `*.graphql` files under a directory into a single `string` file before putting it to a binary like [this](https://github.com/tonyghita/graphql-go-example/blob/master/schema/schema.go). It uses go-bindata, but I want to try packr.

```
 Structure of the directory:
   main.go
   schema/
      query.graphql
      mutation.graphql
      types/
          user.graphql
          location.graphql
```

I want to concatenate the content of all these graphql files into a single string.
```
Use case:
  schema, err := graphql.ParseSchema(**RESULT OF PACKR**, &Resolver{})
```

How do I do this with packr.**v2**? I'm relatively new to all of this.

Thanks for reading!

**Comment by markbates** [comment]
Not sure why you would want to do that, but I’m sure you have your reasons. :)

Packr packs files as they appear on disk. If you want to pack things differently you would need to create your own stores and resolvers. 

https://godoc.org/github.com/gobuffalo/packr/v2/file/resolver#Resolver
https://godoc.org/github.com/gobuffalo/packr/v2/jam/store#Store

Here is a video demonstrating that. https://blog.gobuffalo.io/packr-v2-0-0-rc-3-introduction-7a5e06a688c5

**Comment by akashbdj** [comment]
Hi Mark,

My use case is: I want to split schema definition into multiple files, keeping it more modular and maintainable. But [graphql](https://github.com/graph-gophers/graphql-go) library which I'm using takes `schema` as `string` value as an argument.

**Here's the signature of the graphql lib:**
```go
schema, err := graphql.ParseSchema(STRING_REPRESTATION_OF_ENTIRE_SCHEMA, &Resolver{})
```

And, I want to ship that schema(string) inside binary, which will be generated automatically every time I build the project thereby keeping it up to date.

If I use packr with the existing `store` and `resolver`(as you mentioned), I need to do this:

```go
Assuming this is the directory structure:
main.go
schema/
    schema.graphql
    query.graphql
    mutation.graphql
    types/
        user.graphql
        location.graphql


box := packr.New("SchemaPackr", PATH_TO_SCHEMA_DIRECTORY)
schemaStr, _ := box.FindString("schema.graphql")
queryStr, _ := box.FindString("query.graphql")
mutationStr, _ := box.FindString("mutation.graphql")
typesStr, _ := box.FindString("types/types.graphql")

finalStr :=  schemaStr + queryStr + mutationStr + typesStr
schema, err := graphql.ParseSchema(finalStr, &Resolver{}) 

```
I think this is not the idiomatic way of doing it. Imagine I've 20 files, I don't want to concatenate 20 files here. 

I'm not sure if `packr` is the right tool to do this work. Let me know if writing my own `store` and `resolver` would solve the problem. I'd love to work on that!


Thanks for your time,
Akash

**Comment by markbates** [comment]
You can either use box.Walk to walk through all the files to concat them or you can wrap the box with your own implementation of packd.Box to return the files how you’d like. Or you can write your own store and resolver. Those are basically your options.

-----------
Mark Bates

**Comment by akashbdj** [comment]
Hi Mark,
Thank you so much! It's working now. I've used `box.Walk`, it's simple and clean.

**One doubt here:**
I created the `box` inside a directory which is different from the one where I had put my `.graphql` files. Now when I `walk` through the `box`, it loops over item from the current directory where I created the box instead of the `.graphql` files. To make it more clear, consider this:

```
Directory structure:

main.go
someDir/
    myFile.go    <-------- I created a box here. When I walk over box, it prints `xyz.go` and `abc.go`
    xyz.go
    abc.go
schema/
    schema.graphql
    query.graphql
    mutation.graphql
    types/
        user.graphql
        location.graphql
```

What I expected was it to walk over files under `schema/...` directory.

Code inside `someDir/myFile.go` is:
```go
box := packr.New("SchemaPackr", "../schema")
```

What I thought was, it should pack items from `./schema` directory and should walk over those items only. But it's walking over items of `someDir`

Is this expected? I'm sorry if it's a noob question. 

Thanks once again,
Akash
